sonar-scanner-engine/src/main/java/org/sonar/scanner/deprecated/perspectives/ScannerPerspectives.java

```diff
@@ -52,15 +52,15 @@ public BatchPerspectives(PerspectiveBuilder[] builders, DefaultIndex resourceInd
     }
     if (indexedResource != null) {
       PerspectiveBuilder<P> builder = builderFor(perspectiveClass);
-      return builder.loadPerspective(perspectiveClass, componentCache.get(indexedResource));
+      return builder.loadPerspective(perspectiveClass, componentCache.get(indexedResource).inputComponent());
     }
     return null;
   }
 
   @Override
   public <P extends Perspective> P as(Class<P> perspectiveClass, InputPath inputPath) {
     PerspectiveBuilder<P> builder = builderFor(perspectiveClass);
-    return builder.loadPerspective(perspectiveClass, componentCache.get(inputPath));
+    return builder.loadPerspective(perspectiveClass, inputPath);
   }
 
   private <T extends Perspective> PerspectiveBuilder<T> builderFor(Class<T> clazz) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "Perspective"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)